File: app/schemas/agent.py
"""
Agent 相关 Schema
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class AgentProfileUpdateRequest(BaseModel):
    """Agent 信息更新请求"""
    name: str = Field(..., description="Agent 名称")
    description: Optional[str] = Field(None, description="Agent 描述")
    interests: List[str] = Field(default_factory=list, description="兴趣标签列表")
    autonomy_level: str = Field("80", description="自主程度 (0-100)")


class AgentProfileResponse(BaseModel):
    """Agent 信息响应"""
    agent_id: str
    user_id: str
    name: str
    description: Optional[str]
    interests: List[str]
    autonomy_level: str
    is_active: bool
    last_active_at: Optional[datetime]
    connected_at: datetime


class AgentSearchRequest(BaseModel):
    """Agent 搜索请求"""
    interests: Optional[List[str]] = Field(None, description="兴趣标签")
    keyword: Optional[str] = Field(None, description="关键词")


class SimilarAgentResponse(BaseModel):
    """相似 Agent 响应"""
    agent_id: str
    name: str
    description: Optional[str]
    interests: List[str]
    similarity_score: float = Field(..., ge=0, le=1)

File: frontend/src/types/agent.ts
// Agent 类型定义（匹配后端响应）
export interface Agent {
  agent_id: string;
  user_id: string;
  name: string;
  description: string;
  interests: string[]; // 后端使用 interests 而不是 interest_tags
  autonomy_level: number;
  is_active: boolean;
  last_active_at: string | null;
  connected_at: string | null;
  updated_at: string | null;
  stats?: {
    posts_count: number;
    comments_count: number;
    friends_count: number;
  };
  avatar_url?: string;
  online_status?: 'online' | 'offline';
}

// 用户类型
export interface User {
  user_id: string;
  second_me_user_id: string;
  email: string;
  username: string;
  avatar_url?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}
